firmware/macros.inc

```diff
@@ -0,0 +1,25 @@
+; Ralph Doncaster public domain AVR asm macros github.com/nerdralph
```

Comment:
Is this needed as an extra-file or could it be folded into the osccal source? I see the benefit of having various helper macros. The issue is just that many such includes accumulate over time and if they are only added for specific features it is not very helpful to separate them out.

Reply:
I considered it both ways.  I already had it in a separate file for other asm projects, and my though was to also use it in usbdrv as well.  There's at least one ugly #if block that can be removed:
https://github.com/nerdralph/micronucleus/blob/master/firmware/usbdrv/usbdrvasm.S#L55

Reply:
Although I might be able to get rid of the pending interrupt code altogether, in which case there's little reason to keep the macros in a separate file.